overlord/snapstate/handlers.go

```diff
@@ -1172,7 +1172,27 @@ func (m *SnapManager) doCopySnapData(t *state.Task, _ *tomb.Tomb) error {
 		err := SetTaskSnapSetup(t, snapsup)
 		st.Unlock()
 		if err != nil {
-			return fmt.Errorf("cannot set migration status to done (migration won't be undone if change fails): %w", err)
+			return fmt.Errorf("cannot set migration status to done: %w", err)
+		}
+	} else if !opts.UseHidden && opts.MigratedToHidden {
+		// migration was done but user turned the feature off, so undo migration
+		if err := m.backend.UndoHideSnapData(snapsup.InstanceName()); err != nil {
+			if err := m.backend.HideSnapData(snapsup.InstanceName()); err != nil {
+				st.Lock()
+				t.Logf("cannot revert snap dir unhiding (must manually restore %s's dirs from %s to %s): %v",
+					snapsup.InstanceName(), dirs.HiddenSnapDataHomeDir, dirs.UserHomeSnapDir, err)
```

Comment:
Should we use warning here (i.e. `snap warnings`), otherwise this advice is hard for discover for the user? Or use this message to decorate the main error?

Reply:
Makes sense, thank you! I pushed a change to use Warnf and also tried to make the message a bit better. Let me know what you think